lib/Helper/HtmlToDomParser.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+namespace OCA\Cookbook\Helper;
+
+use DOMDocument;
+use OCP\IL10N;
+use LibXMLError;
+use OCP\ILogger;
+use OCA\Cookbook\Exception\ImportException;
+
+/**
+ * This class allows to parse a HTML document into a DOMDocuemnt.
+ *
+ * Its main purpose is to handle the very verbose libxml extension better.
+ */
+class HtmlToDomParser {
+	/**
+	 * Indicates the parsing was successfully terminated
+	 * @var integer
+	 */
+	public const PARSING_SUCCESS = 0;
+	/**
+	 * Indicates the parsing terminated with warnings
+	 * @var integer
+	 */
+	public const PARSING_WARNING = 1;
+	/**
+	 * Indicates the parsing terminated with an error
+	 * @var integer
+	 */
+	public const PARSING_ERROR = 2;
+	/**
+	 * Indicates that the parsing terminated with a fatal error
+	 * @var integer
+	 */
+	public const PARSING_FATAL_ERROR = 3;
+
+	/**
+	 * @var int
+	 */
+	private $state;
+
+	/**
+	 * @var ILogger
+	 */
+	private $logger;
+
+	/**
+	 * @var IL10N
+	 */
+	private $l;
+
+	public function __construct(ILogger $logger, IL10N $il10n) {
+		$this->logger = $logger;
+		$this->l = $il10n;
+	}
+
+	/**
+	 * Parse a HTML file logging any error in the syntax
+	 *
+	 * @param DOMDocument $dom The dom docuemnt to use. Just give a new document in doubt.
+	 * @param string $url The URL of the parsed recipe
+	 * @param string $html The string representation of the HTML file to parse
+	 * @throws ImportException If the parsing of the HTML page failed completely
+	 * @return DOMDocument The parsed document
+	 */
+	public function loadHtmlString(DOMDocument $dom, string $url, string $html): DOMDocument {
+		$libxml_previous_state = libxml_use_internal_errors(true);
+		
+		try {
+			$parsedSuccessfully = $dom->loadHTML($html);
+			
+			// Error handling
+			$errors = libxml_get_errors();
+			$this->checkXMLErrors($errors, $url);
+			libxml_clear_errors();
+			
+			if (!$parsedSuccessfully) {
+				throw new ImportException($this->l->t('Parsing of HTML failed.'));
+			}
+		} finally {
+			libxml_use_internal_errors($libxml_previous_state);
+		}
+		
+		return $dom;
+	}
+
+	/**
+	 * Get the error state of the last parsing routine.
+	 *
+	 * See the constants in the class for possible values.
+	 *
+	 * @return int
+	 */
+	public function getState() {
+		return $this->state;
+	}
+
+	/**
+	 * Compress the xml errors to small mount of log messages
+	 *
+	 * The return value indicates what the most critical value was.
+	 * The value can be PARSING_SUCCESS, PARSING_WARNING, PARSING_ERROR, or PARSING_FATAL_ERROR.
+	 *
+	 * @param array $errors The array of all parsed errors
+	 * @param string $url The parsed URL
+	 * @return int Indicator what the most severe issue was
```

Comment:
```suggestion
	 * @return int Indicator what the most severe issue was
	 * @throws \Exception
```

Documented return type does not match void return type